examples/3d/meshlet.rs

```diff
@@ -0,0 +1,258 @@
+
+            info!("Calculating dragon meshlets...");
+            *dragon_meshlet_mesh_handle =
+                meshlet_meshes.add(MeshletMesh::from_mesh(dragon_mesh).unwrap());
+            info!("Dragon meshlets calculated");
+
+            commands.spawn(MaterialMeshletMeshBundle {
+                meshlet_mesh: dragon_meshlet_mesh_handle.clone(),
+                material: materials.add(StandardMaterial {
+                    base_color: Color::hex("#ffd891").unwrap(),
+                    metallic: 1.0,
+                    perceptual_roughness: 0.9,
+                    ..default()
+                }),
+                transform: Transform::default().with_rotation(Quat::from_rotation_x(PI / 2.0)),
+                ..default()
+            });
+
+            commands.spawn(MaterialMeshletMeshBundle {
+                meshlet_mesh: dragon_meshlet_mesh_handle.clone(),
+                material: materials.add(StandardMaterial {
+                    base_color: Color::hex("#ffc0cb").unwrap(),
+                    perceptual_roughness: 0.1,
+                    ..default()
+                }),
+                transform: Transform::default()
+                    .with_rotation(Quat::from_rotation_x(PI / 2.0))
+                    .with_translation(Vec3::new(1.0, 0.0, 0.0))
+                    .with_scale(Vec3::splat(0.5)),
+                ..default()
+            });
+        }
+    }
+}
+
+// --------------------------------------------------------------------------------------
+
+use bevy::input::mouse::MouseMotion;
+use bevy::window::CursorGrabMode;
+use std::f32::consts::*;
+
+pub const RADIANS_PER_DOT: f32 = 1.0 / 180.0;
+
+#[derive(Component)]
+pub struct CameraController {
+    pub enabled: bool,
+    pub initialized: bool,
+    pub sensitivity: f32,
+    pub key_forward: KeyCode,
+    pub key_back: KeyCode,
+    pub key_left: KeyCode,
+    pub key_right: KeyCode,
+    pub key_up: KeyCode,
+    pub key_down: KeyCode,
+    pub key_run: KeyCode,
+    pub mouse_key_enable_mouse: MouseButton,
+    pub keyboard_key_enable_mouse: KeyCode,
+    pub walk_speed: f32,
+    pub run_speed: f32,
+    pub friction: f32,
+    pub pitch: f32,
+    pub yaw: f32,
+    pub velocity: Vec3,
+}
+
+impl Default for CameraController {
+    fn default() -> Self {
+        Self {
+            enabled: true,
+            initialized: false,
+            sensitivity: 1.0,
+            key_forward: KeyCode::KeyW,
+            key_back: KeyCode::KeyS,
+            key_left: KeyCode::KeyA,
+            key_right: KeyCode::KeyD,
+            key_up: KeyCode::KeyE,
+            key_down: KeyCode::KeyQ,
+            key_run: KeyCode::ShiftLeft,
+            mouse_key_enable_mouse: MouseButton::Left,
+            keyboard_key_enable_mouse: KeyCode::KeyM,
+            walk_speed: 5.0,
+            run_speed: 15.0,
+            friction: 0.5,
+            pitch: 0.0,
+            yaw: 0.0,
+            velocity: Vec3::ZERO,
+        }
+    }
+}
+
+fn camera_controller(
```

Comment:
don't know if theres a policy for examples to be maximally self contained but it seems that this kinda code could be shared by many examples

Reply:
Yeah it's code copied from the scene viewer example. The whole example needs cleaning up in the future, this is just temporary. It's on the TODO task list.